common/src/VisClient.cpp

```diff
@@ -0,0 +1,825 @@
+    mThread = std::thread([this]() {
+        init();
+        while (mIsActive == true) {
+            poll();
+            std::this_thread::sleep_for(std::chrono::milliseconds(60));
+        }
+        if (mWs) {
+            mWs->close();
+            mWs = nullptr;
+        }
+        {
+            std::lock_guard<std::mutex> lock(mLock);
+            mConnectedState = ConnState::STATE_DISCONNECTED;
+        }
+    });
+    ALOGV("VisClient thread has started ...");
+}
+
+bool VisClient::waitConnectionS(int s) const {
+    ALOGV("waitConnectionMs");
+    int minSleepTime = std::min(s, 1);
+    int timeSleeped = 0;
+    std::unique_lock<std::mutex> lock(mLock);
+    while (mConnectedState != epam::ConnState::STATE_CONNECTED) {
+        lock.unlock();
+        sleep(minSleepTime);
+        lock.lock();
+        timeSleeped += minSleepTime;
+        if (timeSleeped >= s) break;
+    }
+    return mConnectedState == epam::ConnState::STATE_CONNECTED;
+}
+
+void VisClient::stop() {
+    ALOGV("stop");
+    mIsActive = false;
+    if (mThread.joinable()) {
+        mThread.join();
+    }
+    ALOGI("VisClient thread has stopped ...");
+}
+
+int VisClient::poll() {
+    mLock.lock();
+    ALOGV("poll, connection state is %d", static_cast<int>(mConnectedState));
+    if (mConnectedState == ConnState::STATE_DISCONNECTED) {
+        doInit();
+    } else if (mConnectedState == ConnState::STATE_CONNECTED) {
+#ifdef RECOVER_SUBSCRIBE
+        for (auto it = mRecoverObservers.begin(); it != mRecoverObservers.end();) {
+            ALOGI("Found some observer %du for recovery", it->first);
+            int id = sendWSMessage(requestTemplateSubscribe, it->second.propertyName.c_str());
+            if (id < 0) {
+                ALOGE("Unable to recover observer, will do it later %du", it->first);
+                ++it;
+                continue;
+            }
+
+            SubscriptionRequest sr{it->second.handler, it->second.propertyName,
+                                   static_cast<size_t>(id)};
+            mSubscriptionObservers.insert(std::make_pair(id, sr));
+            std::promise<WMessageResult> p;
+            mPromised.emplace(std::pair<int, std::promise<WMessageResult> >(id, std::move(p)));
+            it = mRecoverObservers.erase(it);
+        }
+#endif
+    }
+    mLock.unlock();
+
+    // poll can trigger other event under mLock
+    mHub.poll();
+    return 0;
+}
+
+int VisClient::doInit() {
+    ALOGV("doInit");
+    mRequestId = 0;
+    mProtocolErrorCount = 0;
+    ALOGD("Will try to connect to VIS uri[%s]", mUri.c_str());
+    mHub.connect(mUri.c_str(), reinterpret_cast<void*>(CONNECTION_MODE_SSL));
+    mConnectedState = ConnState::STATE_CONNECTING;
+    return 0;
+}
+
+int VisClient::init() {
+    std::lock_guard<std::mutex> lock(mLock);
+    ALOGV("init");
+
+    if (mUri.empty()) {
+        char propValue[PROPERTY_VALUE_MAX] = {};
+        property_get("persist.vis.uri", propValue, "wss://192.168.1.100:8088");
```

Comment:
Is it expected to be hard coded like this?
Maybe put it as define to top of file with explanation in comments?